compiler/rustc_passes/src/liveness.rs

```diff
@@ -329,6 +329,48 @@ impl<'tcx> Visitor<'tcx> for IrMaps<'tcx> {
             }
         }
 
+        // Allow todo! macro
+        /*
+            Skips checking for unused variables when the trailing expression
+            of the body is a panic with a message that contains "not yet implemented".
+
+            # Example
+
+            fn foo(x: i32) {
+                // arbitrary code
+                todo!()
+            }
+        */
+        if let ExprKind::Block(block, _) = &body.value.kind {
+            if let Some(expr) = block.expr {
+                if let ExprKind::Call(call, [arg]) = expr.kind {
+                    if let ExprKind::Path(QPath::Resolved(_, path)) = call.kind {
+                        if let Res::Def(DefKind::Fn, path_def_id) = &path.res {
+                            let panic_fn = self.tcx.lang_items().panic_fn();
+                            // Note: there is no function for panic_fmt, so we have to extract it from the debug output :(
+                            // builder doesn't like this being called without panic `self.tcx.def_path_str(*path_def_id);`
+                            let path_str = format!("{:?}", path_def_id);
```

Comment:
What does this mean? Why did `self.tcx.def_path_str` not work?

Reply:
It errors with:
    
    thread 'rustc' panicked at 'no warnings or errors encountered even though `delayed_good_path_bugs` issued', compiler/rustc_errors/src/lib.rs:974:13